File: mfiles/topology/boost.m
classdef boost < n_iso_dcdc
  properties
  end
  methods
    function y = D(obj,vin,vout)
      y=(vout-vin)/vout;
    end
    function obj=boost(vimax,vimin,vomax,vomin,pomax)
      obj=obj@n_iso_dcdc(vimax,vimin,vomax,vomin);

      % boost only steps up
      if vimax>=vomin
        error('maximum input > minimum output')
      end


      obj.Dmax=(vomax-vimin)/vomax; % max duty cycle
      obj.Dmin=(vomin-vimax)/vomin; % min duty cycle
      obj.V_Qmain=vomax; % voltage stress on main switch
      obj.V_Qsync=vomax; % voltage stress on synchronous switch
      obj.I_Qsync=(pomax/vomin); % current through synchronous switch
      obj.I_Qmain=obj.I_Qsync.*(1/(1-obj.Dmax)); % current through main switch

    end
  end
end

File: python/topology/boost.py
from converter import converter
# diode reset forward converter
# tertiary winding

class boost(converter):

    #I_Qsync  # current stress synchronous switch on secondary side
    def __init__(self,vimax,vimin,vomax,vomin,pomax,fsw):
        converter.__init__(self,vimax,vimin,vomax,vomin,pomax,fsw)
        '''
        self.Dmin=0.15 # set minimum duty cycle
        self.nps = vimax*self.Dmin/vomin # set turns ratio based on min duty cycle
        self.Dmax = self.nps*vomax/vimin # max duty cycle
        self.V_Qp = vimax+vimax*(self.Dmax/(1-self.Dmax)) # voltage stress on primary
        self.V_Qs = vimax/self.nps  # voltage stress on secondary switches

        self.nsw = 3 # number of active swith not including diode on reset winding
        self.nwinding = 3 # number of winding including reset winding
        self.I_Qp = (pomax/vomin)*self.Dmax/self.nps # current stress on primary switch
        self.I_Qs = (pomax/vomin)*self.Dmax # current stress on secondry switch
        self.I_Qsync=(pomax/vomin)*(1-self.Dmax) # current stress on secondary synchronous switch
        '''
